### PR TITLE
Issue 462 add change and reset password feature

### PR DESCRIPTION
**Pull request by supersoild2000**
develop

## GitHub Board

**Issue link**

[#462 Add change and reset password features ]( https://github.com/ita-social-projects/dokazovi-fe/issues/462 )

**Story link**

[#81 Change password.](https://github.com/ita-social-projects/dokazovi-requirements/issues/81)
[#77 Forgot password](https://github.com/ita-social-projects/dokazovi-requirements/issues/77)

## Summary of issue

- [x]  Implement form to send on the email address link to change password in 'Особистий кабінет' in tab 'Змінити пароль'.
- [x]  Implement form to send on the email address link to reset password in LoginModal component.
- [x]  Implement page for update password.

## Summary of change

- [x]   Implemented form to send on the email address link to change password in 'Особистий кабінет' in tab 'Змінити пароль'.
- [x]   Implemented form to send on the email address link to reset password in LoginModal component.
- [x]   Implement page for update password.
- [x]   Refactored LoginModal component

### CODE REVIEW

File: src/components/Form/Buttons/BasicAcceptButton.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { useStyles } from './styles/BasicButton.style';
+
+interface IBasicAcceptButtonProps {
+  className?: string;
+  disabled?: boolean;
+  label?: string;
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+}
+
+export const BasicAcceptButton: React.FC<IBasicAcceptButtonProps> = ({
```
Comment (michalenr) [inline src/components/Form/Buttons/BasicAcceptButton.tsx:13]: this and BasicSignButton seem to be 100% the same. The only difference is styles. I think they can be merged into one

File: src/old/lib/utilities/API/api.ts [hunk 2]
```diff
@@ -192,6 +192,44 @@ export const login = async (
...
+  newPassword: string,
+  matchPassword: string,
+): Promise<boolean> => {
+  try {
+    const response = await instance.post('/user/update-password', {
+      token,
+      newPassword,
+      matchPassword,
+    });
+    return response.status === 200;
```
Comment (michalenr) [inline src/old/lib/utilities/API/api.ts:218]: what if response is not 200, but within 200-299 range? 

Better to check like this:

```suggestion
    return response.ok;
```

File: src/old/lib/utilities/API/api.ts [hunk 3]
```diff
@@ -192,6 +192,44 @@ export const login = async (
...
+  } catch {
+    return false;
+  }
+};
+
+export const checkPasswordToken = async (token: string): Promise<boolean> => {
+  try {
+    const response = await instance.get(`/user/check-token?token=${token}`);
+    return response.status === 200;
```
Comment (michalenr) [inline src/old/lib/utilities/API/api.ts:227]: same as above

File: src/views/PasswordUpdate/PasswordUpdateView.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  newPassword: string;
+  matchPassword: string;
+}
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+const PasswordUpdateView = () => {
+  const query = useQuery();
+  const history = useHistory();
+  const classes = useStyles();
+  const [token] = useState<string | null>(query.get('token'));
```
Comment (michalenr) [inline src/views/PasswordUpdate/PasswordUpdateView.tsx:29]: why do we need to put it in useState?

File: src/views/PasswordUpdate/PasswordUpdateView.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        disabled={
+          !formState.isValid ||
+          watch('newPassword')?.length === 0 ||
+          watch('matchPassword')?.length === 0
+        }
+      />
+    </form>
+  );
+  if (doesTokenValid) {
+    if (success) {
```
Comment (michalenr) [inline src/views/PasswordUpdate/PasswordUpdateView.tsx:126]: In my opinion, a ternary operator here would be more readable than nested if's

```js
return success ? newPasswordUpdated : newPasswordForm
```

**Review by Valentyna-Chudik — APPROVED** [review]

**Review by sarhan-azizov — APPROVED** [review]

**Pull request merged**